dcc-import-gene/src/main/java/org/icgc/dcc/imports/gene/reader/ExternalReader.java

```diff
@@ -0,0 +1,119 @@
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import lombok.val;
+
+/**
+ * Responsible for creating a map of gene to external db ids
+ */
+public final class ExternalReader {
+
+  /**
+   * Constants
+   */
+  private static final String OBJECT_XREF_URI =
+      "ftp://ftp.ensembl.org/pub/grch37/release-82/mysql/homo_sapiens_core_82_37/object_xref.txt.gz";
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  public static Map<String, ObjectNode> externalIds() {
+    val geneIdMap = GeneReader.geneIdMap();
+    val transToGeneMap = TransReader.translationToGene();
+
+    val retMap = new HashMap<String, ObjectNode>();
+    BaseReader.read(OBJECT_XREF_URI, line -> {
+      if ("Gene".equals(line[2])) {
+        String geneId = geneIdMap.get(line[1]);
+
+        ObjectNode externalDbs;
+        if (retMap.containsKey(geneId)) {
+          externalDbs = retMap.get(geneId);
+        } else {
+          externalDbs = MAPPER.createObjectNode();
+          externalDbs.put("entrez_gene", MAPPER.createArrayNode());
+          externalDbs.put("hgnc", MAPPER.createArrayNode());
+          externalDbs.put("omim_gene", MAPPER.createArrayNode());
+          externalDbs.put("uniprotkb_swissprot", MAPPER.createArrayNode());
+          retMap.put(geneId, externalDbs);
+        }
+
+        String xrefId = line[3];
+        if (NameReader.entrezMap.containsKey(xrefId)) {
+          ArrayNode arrayNode = (ArrayNode) externalDbs.get("entrez_gene");
+          arrayNode.add(NameReader.entrezMap.get(xrefId));
+
+        } else if (NameReader.hgncMap.containsKey(xrefId)) {
+          ArrayNode arrayNode = (ArrayNode) externalDbs.get("hgnc");
+          arrayNode.add(NameReader.hgncMap.get(xrefId));
+
+        } else if (NameReader.mimMap.containsKey(xrefId)) {
+          ArrayNode arrayNode = (ArrayNode) externalDbs.get("omim_gene");
+          arrayNode.add(NameReader.mimMap.get(xrefId));
+        }
+
+      } else if ("Translation".equals(line[2])) {
+        // Uniprot Ids are for proteins, which means we match them to translations and eventually work up to a gene/
+
+        String geneId = geneIdMap.get(transToGeneMap.get(line[1]));
+
+        ObjectNode externalDbs;
+        if (retMap.containsKey(geneId)) {
+          externalDbs = retMap.get(geneId);
+        } else {
+          externalDbs = MAPPER.createObjectNode();
+          externalDbs.put("entrez_gene", MAPPER.createArrayNode());
+          externalDbs.put("hgnc", MAPPER.createArrayNode());
+          externalDbs.put("omim_gene", MAPPER.createArrayNode());
+          externalDbs.put("uniprotkb_swissprot", MAPPER.createArrayNode());
+          retMap.put(geneId, externalDbs);
+        }
+
+        String xrefId = line[3];
+        if (NameReader.uniprotMap.containsKey(xrefId)) {
+          ArrayNode arrayNode = (ArrayNode) externalDbs.get("uniprotkb_swissprot");
+          String uniprot = NameReader.uniprotMap.get(xrefId);
```

Comment:
We should avoid using statics like this as it makes testing more difficult because there is an implicit ordering that isn't enforced by a constructor. My opinion is that this class should have its dependencies passed to it.